v03_pipeline/lib/tasks/base/base_variant_annotations_table.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+import hail as hl
+import luigi
+
+from v03_pipeline.lib.annotations.fields import get_fields
+from v03_pipeline.lib.model import AnnotationType
+from v03_pipeline.lib.paths import (
+    reference_dataset_collection_path,
+    variant_annotations_table_path,
+)
+from v03_pipeline.lib.tasks.base.base_pipeline_task import BasePipelineTask
+from v03_pipeline.lib.tasks.files import (
+    GCSorLocalFolderTarget,
+    GCSorLocalTarget,
+    HailTableTask,
+)
+
+
+class BaseVariantAnnotationsTableTask(BasePipelineTask):
```

Comment:
Both `UpdateVariantAnnotationsTableWithNewSamplesTask` and the (not yet written) `UpdateVariantAnnotationsTableWithNewReferenceDatasetTask` will share this code.